kyverno-1.13 fails to build on amd64

The package kyverno-1.13 with version 1.13.4 and epoch r0 fails to build from source with the following error:

```
[git checkout] execute: git fetch --quiet origin --depth=1 --no-tags +refs/tags/v1.13.4:refs/origin/tags/v1.13.4
[git checkout] execute: git checkout --quiet origin/tags/v1.13.4
[git checkout] tag v1.13.4 is c6e3bcddee8648b95ab717e658e541809705f9bb
running step "go/bump"
2025/02/19 07:09:40 Running go mod tidy with go version '1.24.0' ...
2025/02/19 07:09:53 Update package: github.com/go-git/go-git/v5
2025/02/19 07:09:53 Running go mod edit -droprequire ...
2025/02/19 07:09:53 Running go get ...
2025/02/19 07:09:54 Running go mod tidy with go version '1.24.0' ...
running step "patch"
Hmm...  Looks like a unified diff to me...
The text leading up to this was:
--------------------------
|From f68f4c5a8f9566613042b6a74d7b8fbf8c5c8bd4 Mon Sep 17 00:00:00 2001
|From: Eoghan Conlon O'Neill <eoghan.conlononeill@chainguard.dev>
|Date: Wed, 25 Sep 2024 14:20:27 -0600
|Subject: [PATCH] Update otel semconv to 1.26.0
|
|---
| pkg/metrics/metrics.go           |  2 +-
| pkg/tracing/config.go            |  2 +-
| pkg/tracing/helpers.go           |  4 ++--
| pkg/webhooks/handlers/metrics.go |  6 +++---
| pkg/webhooks/handlers/trace.go   | 11 ++++++-----
| 5 files changed, 13 insertions(+), 12 deletions(-)
|
|diff --git a/pkg/metrics/metrics.go b/pkg/metrics/metrics.go
|index 7498248e5..89016b8e0 100644
|--- a/pkg/metrics/metrics.go
|+++ b/pkg/metrics/metrics.go
--------------------------
patching file pkg/metrics/metrics.go
Using Plan A...
Hunk #1 succeeded at 18.
Hmm...  The next patch looks like a unified diff to me...
The text leading up to this was:
--------------------------
|diff --git a/pkg/tracing/config.go b/pkg/tracing/config.go
|index f9a5b6c18..919290970 100644
|--- a/pkg/tracing/config.go
|+++ b/pkg/tracing/config.go
--------------------------
patching file pkg/tracing/config.go
Using Plan A...
Hunk #1 succeeded at 13.
Hmm...  The next patch looks like a unified diff to me...
The text leading up to this was:
--------------------------
|diff --git a/pkg/tracing/helpers.go b/pkg/tracing/helpers.go
|index 57ff265be..a96fb8e20 100644
|--- a/pkg/tracing/helpers.go
|+++ b/pkg/tracing/helpers.go
--------------------------
patching file pkg/tracing/helpers.go
Using Plan A...
Hunk #1 succeeded at 6.
Hunk #2 succeeded at 28.
Hmm...  The next patch looks like a unified diff to me...
The text leading up to this was:
--------------------------
|diff --git a/pkg/webhooks/handlers/metrics.go b/pkg/webhooks/handlers/metrics.go
|index bc10fab7b..0128d1036 100644
|--- a/pkg/webhooks/handlers/metrics.go
|+++ b/pkg/webhooks/handlers/metrics.go
--------------------------
patching file pkg/webhooks/handlers/metrics.go
Using Plan A...
Hunk #1 succeeded at 12.
Hunk #2 succeeded at 86.
Hmm...  The next patch looks like a unified diff to me...
The text leading up to this was:
--------------------------
|diff --git a/pkg/webhooks/handlers/trace.go b/pkg/webhooks/handlers/trace.go
|index 6083bbe02..350807141 100644
|--- a/pkg/webhooks/handlers/trace.go
|+++ b/pkg/webhooks/handlers/trace.go
--------------------------
patching file pkg/webhooks/handlers/trace.go
Using Plan A...
Hunk #1 succeeded at 9.
Hunk #2 succeeded at 24.
Hmm...  Ignoring the trailing garbage.
done
Go fmt...
Go vet...
# github.com/kyverno/kyverno/pkg/engine/utils
# [github.com/kyverno/kyverno/pkg/engine/utils]
pkg/engine/utils/match.go:247:21: non-constant format string in call to fmt.Errorf
# github.com/kyverno/kyverno/pkg/policy/generate
# [github.com/kyverno/kyverno/pkg/policy/generate]
pkg/policy/generate/validate.go:113:21: non-constant format string in call to fmt.Errorf
# github.com/kyverno/kyverno/pkg/policy/mutate
pkg/policy/mutate/validate.go:126:35: non-constant format string in call to fmt.Errorf
# github.com/kyverno/kyverno/pkg/webhooks/resource/mutation
pkg/webhooks/resource/mutation/mutation.go:222:27: non-constant format string in call to fmt.Errorf
# github.com/kyverno/kyverno/cmd/cli/kubectl-kyverno/processor
# [github.com/kyverno/kyverno/cmd/cli/kubectl-kyverno/processor]
cmd/cli/kubectl-kyverno/processor/policy_processor.go:386:23: non-constant format string in call to fmt.Fprintf
make: *** [Makefile:182: vet] Error 1
failed to build package: unable to run package kyverno-1.13 pipeline: unable to run pipeline: exit status 2
```

--- a/kyverno-1.13.yaml
+++ b/kyverno-1.13.yaml
@@ -1,7 +1,7 @@
 package:
   name: kyverno-1.13
   version: "1.13.4"
-  epoch: 0
+  epoch: 1
   description: Kubernetes Native Policy Management
   copyright:
     - license: Apache-2.0
@@ -42,7 +42,7 @@ pipeline:
   - uses: patch
     # This patch (ideally) can be removed when a new release is cut by the kyverno maintainers.
     with:
-      patches: update-otel-semconv-to-1.26.0.patch
+      patches: update-otel-semconv-to-1.26.0.patch disable-vet-printf-checks.patch
 
   - runs: |
       make build-all

--- a/kyverno-1.13/disable-vet-printf-checks.patch
+++ b/kyverno-1.13/disable-vet-printf-checks.patch
@@ -1,0 +1,17 @@
+Go 1.24 introduces a new vet check ("non-constant format string in call to
+fmt.Errorf") which upstream haven't yet handled, and `go vet` is run as part of
+the build: disable the class of checks which are breaking the build.
+---
+diff --git a/Makefile b/Makefile
+index 8227ac8..add1494 100644
+--- a/Makefile
++++ b/Makefile
+@@ -179,7 +179,7 @@ fmt: ## Run go fmt
+ .PHONY: vet
+ vet: ## Run go vet
+ 	@echo Go vet... >&2
+-	@go vet ./...
++	@go vet -printf=false ./...
+ 
+ .PHONY: imports
+ imports: $(GOIMPORTS)

